client/commonplugins/commonplugins.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package commonplugins
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/go-version"
+)
+
+const (
+	CPI_TIMEOUT_SOFT time.Duration = 2 * time.Second
+	CPI_TIMEOUT_HARD time.Duration = 1 * time.Second
```

Comment:
These timeouts are for two different windows: one is the timeout for the command itself (which I think is too small for external API calls) and the other is the timeout for killing the command. I'd probably call these the "command timeout" and the "kill timeout".